app/experimenter/kinto/tasks.py

```diff
@@ -78,54 +78,41 @@ def nimbus_check_kinto_push_queue():
     for application, collection in NimbusExperiment.KINTO_APPLICATION_COLLECTION.items():
         kinto_client = KintoClient(collection)
 
-        rejected_collection_data = kinto_client.get_rejected_collection_data()
-        if rejected_collection_data:
-            rejected_slug = kinto_client.get_rejected_record()
-            experiment = NimbusExperiment.objects.get(slug=rejected_slug)
-            if (
-                experiment.status == NimbusExperiment.Status.LIVE
-                and experiment.is_end_requested
-            ):
-                experiment.is_end_requested = False
-            else:
-                experiment.status = NimbusExperiment.Status.DRAFT
+        if kinto_client.has_pending_review():
+            continue
+        elif kinto_client.has_rejection():
```

Comment:
This should be its own if clause, that way we can pull the rejection out of the collection AND send a new change to it in the same task invocation.